### ISSUE TITLE
Eliminate overlaySSR files where possible

### ISSUE THREAD

**Issue by derimmer** [vasl-developers/vasl#836]
After testing, I am pretty sure that the overlaySSR file is redundant if the boardmetadata.xml file exists. I am going to start updating boards by deleting the overlaySSR file (after having confirmed that the boardmetadata.xml file is present and contains the necessary overlay and underlay rules).  I am will do them in small batches and we should hear from users pretty quickly. 

Once done, we can delete the files from the Github repo and change instructions for setting up bdXX files.

There are probably other files that we can delete (colorSSR is the logical one but it will take more work to ensure that we do it safely).

**Comment by derimmer** [comment]
In board LFT2, I have replaced the OverlaySSR file by adding a BoardMetaData.xml file to the board and adding all of the overlay transforms to it. There are two that don't seem to be working. I can't get them to work in the current (unchanged) version of the board either. 

The first one is CragsToPalmTrees, the second one is Winter (which adds an overlay called LFT2_WinterGully.gif). Would somebody  be kind enough to try these out on their existing version of the board and let me know if they can get them to work or not?

There are about 25 TPP and HASL boards that use the OverlaySSR file but don't have a BoardMetaData.xml file. If I get the LFT2 board to work then I can move to these other boards. That should completely eliminate the use of the OverlaySSR file (one or two minor exceptions) which starts to streamline the terrain transformation process and make it easier to debug problems and add new transforms. 

Thanks.

**Comment by BigAl737** [comment]
Hi Doug, thanks for all our hard work on this. I'd be happy to help here but my computer time is very limited at present. If someone doesn't beat me to it, I'll get to this as soon as I'm able. It'll be at least a week.

**Comment by derimmer** [comment]
> The first one is CragsToPalmTrees, the second one is Winter (which adds an overlay called LFT2_WinterGully.gif). Would somebody be kind enough to try these out on their existing version of the board and let me know if they can get them to work or not?
> 

I fixed the first one. It should have been "CragstoPalm" and not "CragsToPalmTrees". Which points to why we need to do this work. A lot of terrain transform bugs are due to this kind of inconsistency. 

I point no fingers. This is exactly the kind of error I am liable to make. We have too many places where we can make such errors. Getting rid of the OverlaySSR files reduces the risk as will eliminating the other special files in each board. Will take time but worth doing.

**Comment by derimmer** [comment]
> > The first one is CragsToPalmTrees, the second one is Winter (which adds an overlay called LFT2_WinterGully.gif). Would somebody be kind enough to try these out on their existing version of the board and let me know if they can get them to work or not?
> 
> I fixed the first one. It should have been "CragstoPalm" and not "CragsToPalmTrees". Which points to why we need to do this work. A lot of terrain transform bugs are due to this kind of inconsistency.
> 
> I point no fingers. This is exactly the kind of error I am liable to make. We have too many places where we can make such errors. Getting rid of the OverlaySSR files reduces the risk as will eliminating the other special files in each board. Will take time but worth doing.

I have now fixed the WinterGully issue so LFT2 is updated and has been released from captivity.

**Comment by derimmer** [comment]
I have completed this work. I am sure that there are probably some boards in the repo that I have missed but they can be dealt with as need arises. 

At present, all boards will work whether they contain the old OverlaySSR text file or whether they incorporate it into the BoardMetaData.xml file. At some point, as I make code changes that will stop. And we will see if any terrain transformations stop working on specific boards. One thing I have learned is that OverlaySSR files often contain references to terrain transformations that use underlays, such as WallsOnly or WallToCactus. These do not need to be in either the BoardMetaData.xml file or the OverlaySSR file. They are all listed in the SharedBoardMetaData.xml file that is built into VASL and will be triggered by the TerrainSSR dialog when selected. I have tested this on a number of boards and all evidence confirms this. OverlaySSRules appear to be needed only when there is a board specific overlay in a board file that needs to be referenced. The LightWoods terrain transformation works this way so all boards wishing to apply that transformation need an overlay .gif in the board file and an OverlaySSRule in the board's BoardMetaData.xml file.
